mlflow/api/model.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+)
+
+// MLFlowModelAPI defines the response object from the API
+type Model struct {
+	Name                 string   `json:"name"`
+	CreationTimestamp    int64    `json:"creation_timestamp,omitempty"`
+	LastUpdatedTimestamp int64    `json:"last_updated_timestamp,omitempty"`
+	UserID               string   `json:"user_id,omitempty"`
+	LatestVersions       []string `json:"latest_versions,omitempty"`
+	Description          string   `json:"description,omitempty"`
+	Tags                 []Tag    `json:"tags,omitempty"`
+}
+
+type RegisteredModel struct {
+	RegisteredModel Model `json:"registered_model"`
+}
+
+// ModelAPI ...
+type ModelAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+// NewModelAPI ...
+func NewModelAPI(ctx context.Context, m interface{}) ModelAPI {
+	return ModelAPI{m.(*common.DatabricksClient), ctx}
+}
+
+// Create ...
+func (a ModelAPI) Create(d *Model) error {
+	return a.client.Post(a.context, "/mlflow/registered-models/create", d, &d)
+}
+
+// Read ...
+func (a ModelAPI) Read(modelName string) (*Model, error) {
+	//var d MLFLowModelAPI
+	var d RegisteredModel
+	// need to figure out how to send param
+	err := a.client.Get(a.context, fmt.Sprintf("/mlflow/registered-models/get?name=%s", modelName), nil, &d)
```

Comment:
supply a map, not string builder - it should be converted to query string for GET method